lib/Sema/TypeCheckPattern.cpp

```diff
@@ -738,9 +738,14 @@ static bool validateParameterType(ParamDecl *decl, DeclContext *DC,
                                   TypeChecker &TC) {
   if (auto ty = decl->getTypeLoc().getType())
     return ty->is<ErrorType>();
-  
+
+  // If the element is a variadic parameter, resolve the parameter type as if
+  // it were in non-parameter position, since we want functions to be
+  // @escaping in this case.
```

Comment:
On the next line we're choosing whether to pass FunctionInput or not, but that doesn't seem like the right question—this *is* a FunctionInput, isn't it? Just not an immediate-function-input.